src/pretix/base/logentrytypes.py

```diff
@@ -0,0 +1,252 @@
+                    'organizer': event.organizer.slug,
+                    'event': event.slug,
+                }) + '#plugin_' + plugin_name,
+            })
+        else:
+            a_map['val'] = '<i>{val}</i> <span data-toggle="tooltip" title="{errmes}" class="fa fa-warning fa-fw"></span>'.format_map({
+                **a_map,
+                "errmes": _("The relevant plugin is currently not active."),
+            })
+        return wrapper.format_map(a_map)
+
+
+class LogEntryTypeRegistry(EventPluginRegistry):
+    def __init__(self):
+        super().__init__({'action_type': lambda o: getattr(o, 'action_type')})
+
+    def register(self, *objs):
+        for obj in objs:
+            if not isinstance(obj, LogEntryType):
+                raise TypeError('Entries must be derived from LogEntryType')
+
+            if obj.__module__ == LogEntryType.__module__:
+                raise TypeError('Must not register base classes, only derived ones')
+
+        return super().register(*objs)
+
+    def new_from_dict(self, data):
+        """
+        Register multiple instance of a `LogEntryType` class with different `action_type`
+        and plain text strings, as given by the items of the specified data dictionary.
+
+        This method is designed to be used as a decorator as follows:
+
+        .. code-block:: python
+
+            @log_entry_types.new_from_dict({
+                'pretix.event.item.added': _('The product has been created.'),
+                'pretix.event.item.changed': _('The product has been changed.'),
+                # ...
+            })
+            class CoreItemLogEntryType(ItemLogEntryType):
+                # ...
+
+        :param data: action types and descriptions
+                     ``{"some_action_type": "Plain text description", ...}``
+        """
+        def reg(clz):
+            for action_type, plain in data.items():
+                self.register(clz(action_type=action_type, plain=plain))
+            return clz
+        return reg
+
+
+"""
+Registry for LogEntry types.
+
+Each entry in this registry should be an instance of a subclass of ``LogEntryType``.
+They are annotated with their ``action_type`` and the defining ``plugin``.
+"""
+log_entry_types = LogEntryTypeRegistry()
+
+
+class LogEntryType:
+    """
+    Base class for a type of LogEntry, identified by its action_type.
+    """
+
+    def __init__(self, action_type=None, plain=None):
+        if action_type:
+            self.action_type = action_type
+        if plain:
+            self.plain = plain
+
+    def display(self, logentry):
+        """
+        Returns the message to be displayed for a given logentry of this type.
+
+        :return: `str` or `LazyI18nString`
+        """
+        if hasattr(self, 'plain'):
+            plain = str(self.plain)
+            if '{' in plain:
+                data = defaultdict(lambda: '?', logentry.parsed_data)
+                return plain.format_map(data)
```

Comment:
Should we catch any formatting errors? `plain` is coming from translation, so might contain a typo or somehow else contain a misconfigured or mismatching {}. Otherwise it might fail hard. As a fallback just return plain?